saleor/graphql/product/tests/test_category_filtering.py

```diff
@@ -66,3 +68,388 @@ def test_categories_with_filtering(
     content = get_graphql_content(response)
     categories_nodes = content["data"]["categories"]["edges"]
     assert len(categories_nodes) == categories_count
+
+
+GET_FILTERED_PRODUCTS_CATEGORY_QUERY = """
+    query ($channel: String, $filters: ProductFilterInput){
+      categories(first: 5) {
+        edges {
+          node {
+            id
+            name
+            products(first: 5, channel: $channel, filter: $filters) {
+              edges {
+                node {
+                  id
+                  name
+                  channel
+                  attributes {
+                    attribute {
+                      choices(first: 10) {
+                        edges {
+                          node {
+                            slug
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+
+def test_category_filter_products_by_channel(
+    user_api_client,
+    product_list,
+    channel_USD,
+    channel_PLN,
+):
+    ProductChannelListing.objects.filter(
+        product=product_list[0],
+    ).update(channel=channel_PLN)
+
+    variables = {
+        "channel": channel_USD.slug,
+        "filters": {"channel": channel_PLN.slug},
+    }
+
+    response = user_api_client.post_graphql(
+        GET_FILTERED_PRODUCTS_CATEGORY_QUERY, variables
+    )
+
+    content = get_graphql_content(response)
+    for edge in content["data"]["categories"]["edges"]:
+        for inner_edge in edge["node"]["products"]["edges"]:
+            assert inner_edge["node"]["channel"] == channel_USD.slug
+
+
+def test_category_filter_products_by_is_published(
+    user_api_client,
+    product_list_published,
+    channel_USD,
+):
+    """
+    Products created with published collections and expect to see no products after
+    filter.
+    """
+
+    variables = {
+        "channel": channel_USD.slug,
+        "filters": {"isPublished": False},
+    }
+
+    response = user_api_client.post_graphql(
+        GET_FILTERED_PRODUCTS_CATEGORY_QUERY, variables
+    )
+
+    content = get_graphql_content(response)
+    for edge in content["data"]["categories"]["edges"]:
+        assert edge["node"]["products"]["edges"] == []
```

Comment:
I would suggest to modify the test to return some unpublished product, just to confirm, that we return something different that empty list.

Reply:
Test was modified to include unpublished product case as well.